// css-modules-require-hook/dist/validate.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports['default'] = validate;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _lodashForeach = require('lodash.foreach');

var _lodashForeach2 = _interopRequireDefault(_lodashForeach);

var _utility = require('./utility');

var _util = require('util');

var rules = {
  // hook
  extensions: 'array',
  preprocessCss: 'function',
  processCss: 'function',
  to: 'string',
  // plugins
  append: 'array',
  prepend: 'array',
  use: 'array',
  createImportedName: 'function',
  generateScopedName: 'function|string',
  mode: 'string',
  rootDir: 'string'
};

function validate() {
  var options = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];

  (0, _lodashForeach2['default'])(rules, function (types, rule) {
    if (!options[rule]) {
      return;
    }

    if (!types.split('|').some(function (type) {
      return (0, _utility.is)(type, options[rule]);
    })) {
      throw new Error((0, _util.format)('should specify %s for the %s', types, rule));
    }
  });
}

module.exports = exports['default'];
